Bottom: 926b018a0c9a8483c86888bffe3e061f53ffb7f6
Top:    76117077ebd9ada8f4a1b61d2ba78e05914d6b0c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-15 00:09:17 +0000

Refresh of make-rats-snap-to-the-center-o

---

diff --git a/src/find.c b/src/find.c
index dba0e3a..1962234 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2308,15 +2308,19 @@ LOCtoPadRat_callback (const BoxType * b, void *cl)
   if (!TEST_FLAG (TheFlag, rat))
     {
       if (rat->group1 == i->layer &&
-          rat->Point1.X == (i->pad.Point1.X + i->pad.Point2.X) / 2 &&
-          rat->Point1.Y == (i->pad.Point1.Y + i->pad.Point2.Y) / 2)
+	  ((rat->Point1.X == i->pad.Point1.X && rat->Point1.Y == i->pad.Point1.Y) ||
+	   (rat->Point1.X == i->pad.Point2.X && rat->Point1.Y == i->pad.Point2.Y) ||
+	   (rat->Point1.X == (i->pad.Point1.X + i->pad.Point2.X) / 2 &&
+	    rat->Point1.Y == (i->pad.Point1.Y + i->pad.Point2.Y) / 2)))
         {
           if (ADD_RAT_TO_LIST (rat))
             longjmp (i->env, 1);
         }
       else if (rat->group2 == i->layer &&
-               rat->Point2.X == (i->pad.Point1.X + i->pad.Point2.X) / 2 &&
-               rat->Point2.Y == (i->pad.Point1.Y + i->pad.Point2.Y) / 2)
+	       ((rat->Point2.X == i->pad.Point1.X && rat->Point2.Y == i->pad.Point1.Y) ||
+		(rat->Point2.X == i->pad.Point2.X && rat->Point2.Y == i->pad.Point2.Y) ||
+		(rat->Point2.X == (i->pad.Point1.X + i->pad.Point2.X) / 2 &&
+		 rat->Point2.Y == (i->pad.Point1.Y + i->pad.Point2.Y) / 2)))
         {
           if (ADD_RAT_TO_LIST (rat))
             longjmp (i->env, 1);
